pkg/odo/cli/application/application.go

```diff
@@ -76,19 +77,20 @@ func printDeleteAppInfo(client *occlient.Client, appName string, projectName str
 		return errors.Wrap(err, "failed to get Component list")
 	}
 
-	for _, currentComponent := range componentList {
-		componentDesc, err := component.GetComponentDesc(client, currentComponent.ComponentName, appName, projectName)
+	for _, currentComponent := range componentList.Items {
+		componentDesc, err := component.GetComponent(client, currentComponent.Name, appName, projectName)
 		if err != nil {
 			return errors.Wrap(err, "unable to get component description")
 		}
-		log.Info("Component", currentComponent.ComponentName, "will be deleted.")
+		log.Info("Component", currentComponent.Name, "will be deleted.")
 
-		if len(componentDesc.URLs.Items) != 0 {
+		if len(componentDesc.Spec.URL) != 0 {
 			fmt.Println("  Externally exposed URLs will be removed")
 		}
 
-		for _, store := range componentDesc.Storage.Items {
-			fmt.Println("  Storage", store.Name, "of size", store.Spec.Size, "will be removed")
+		for _, store := range componentDesc.Spec.Storage {
+			storo := storage.GetStorage(client, store, currentComponent.Name, appName)
```

Comment:
I guess we can use the ListMounted() method in the pkg/storage package and then iterate over the storage. It will save us extra API calls. Or am I missing something?

Reply:
it's one and the same, instead of writing there, I created separate function which can be reused in future. For example, this function can serve like GET request.